gtsfm/multi_view_optimizer.py

```diff
@@ -50,48 +54,74 @@ def create_computation_graph(
             i2Ui1_graph: relative unit-translations for image pairs, each value wrapped up as Delayed.
             v_corr_idxs_graph: indices of verified correspondences for image pairs, wrapped up as Delayed.
             intrinsics_graph: intrinsics for images, wrapped up as Delayed.
+            two_view_reports_dict: Dict of TwoViewEstimationReports after inlier support processor.
             gt_cameras_graph: list of GT cameras (if they exist), ordered by camera index, wrapped up as Delayed.
 
         Returns:
             The GtsfmData input to bundle adjustment, aligned to GT (if provided), wrapped up as Delayed.
             The final output GtsfmData, wrapped up as Delayed.
+            Dict of TwoViewEstimationReports after view graph estimation.
             List of GtsfmMetricGroups from different modules, wrapped up as Delayed.
         """
-        # prune the graph to a single connected component.
-        pruned_graph = dask.delayed(graph_utils.prune_to_largest_connected_component)(i2Ri1_graph, i2Ui1_graph)
+        view_graph, view_graph_metrics = self.view_graph_estimator.create_computation_graph(
+            i2Ri1_graph,
+            i2Ui1_graph,
+            intrinsics_graph,
+            v_corr_idxs_graph,
+            keypoints_graph,
+            gt_cameras_graph,
+        )
+        view_graph_two_view_report = dask.delayed(filter_dict_keys)(
+            two_view_reports_dict, dask.delayed(lambda x: x.i2Ri1)(view_graph)
+        )
 
-        pruned_i2Ri1_graph = pruned_graph[0]
-        pruned_i2Ui1_graph = pruned_graph[1]
+        # prune the graph to a single connected component.
+        pruned_i2Ri1_graph, pruned_i2Ui1_graph = dask.delayed(graph_utils.prune_to_largest_connected_component, nout=2)(
+            dask.delayed(lambda x: x.i2Ri1)(view_graph), dask.delayed(lambda x: x.i2Ui1)(view_graph)
```

Comment:
@akshay-krishnan what's the advantage of returning a `ViewGraph` type? It looks like we have to use these `dask.delayed()` calls to access its attributes, which seems less clean than just returning 3 return args